libs/entitlement/bridge/src/main/java/org/elasticsearch/entitlement/bridge/package-info.java

```diff
@@ -0,0 +1,70 @@
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+/**
+ * <p>
+ * Contains classes that need to be used directly from instrumented methods.
+ * It's a minimal shim that is patched into the {@code java.base} module so that it is callable from the class library methods instrumented
+ * by the agent. The shim retains a {@link org.elasticsearch.entitlement.bridge.EntitlementChecker} instance (inside its
+ * {@link org.elasticsearch.entitlement.bridge.EntitlementCheckerHandle} holder) and forwards the entitlement checks to the main library,
+ * which is loaded normally.
+ * {@link org.elasticsearch.entitlement.bridge.EntitlementChecker} holds all the entitlements check definitions, one for each instrumented
+ * method.
+ * </p>
+ * <p>
+ * <strong>NOTE:</strong> a subtle point that is worth noting: the {@link org.elasticsearch.entitlement.bridge.EntitlementChecker}
+ * interface needs to be accessible from instrumented methods, hence the need to define it here in this project. But the reference held
+ * by {@link org.elasticsearch.entitlement.bridge.EntitlementCheckerHandle} is an instance of a concrete class that need not be accessible.
+ * Once things get underway, the instrumented code is directly calling a method of {@code ElasticsearchEntitlementChecker}, a class that
+ * it could not resolve, by referencing that method through the {@link org.elasticsearch.entitlement.bridge.EntitlementChecker} interface.
+ * </p>
+ * <p>
+ * In order to work across multiple Java versions, this project uses multi-release jars via the {@code mrjar} plugin, which makes it is
+ * possible to specify classes for specific Java versions in specific {@code src} folders (e.g. {@code main23} for classes available to
+ * Java 23+).
+ * We use the mrjar plugin in a particular way, merging all the classes into the  same jar. Therefore, we want to prefix the file name
+ * with the version, e.g. {@code Java23EntitlementCheckerHandle} and {@code Java23EntitlementChecker}.
+ * </p>
+ * <p>
+ * At runtime, we identify and instantiate the correct class using the runtime Java version to prepend the correct prefix to the class
+ * name  (see {@code EntitlementInitialization#getVersionSpecificCheckerClass}).
+ * </p>
+ * <p>
+ * These different classes are needed to hold entitlements check definitions that are specific to a Java version.
+ * As an example, consider the Linker API.
+ * </p>
+ * <p>
+ * <strong>Note:</strong> the current version of Elasticsearch supports Java 21+; this is only an example (taken from the 8.x branch) that
+ * illustrates a complex scenario.
+ * </p>
+ * <p>
+ * The API went through multiple previews, and therefore changes between Java 19, 20 and 21; in order to support this correctly on these
+ * versions, we should introduce 2 utility interfaces, "preview" and "stable".
+ * For example, for the Java 20 specific signatures and functions, we would create {@code Java20StableEntitlementChecker} and
```

Comment:
These are a lot of implementation details that have nothing to do with the bridge. It seems most of this package info could be moved to the runtime package?

Reply:
Maybe, but these classes are defined here... really not sure where the right place is

Reply:
Moved to the runtime package, it makes more sense